### DISCUSSION TITLE
Configure the streamdeck without the serial number

### DISCUSSION THREAD

**Discussion by flcou** [streamdeck-linux-gui/streamdeck-linux-gui#35] [Ideas / Feature Requests]
For now I have to specify the streamdeck serial number in the configuration.

Would it be possible to make the streamdeck configuration apply to any streamdeck connected ?

**Comment by coolapso** [comment]
What would happen if you have more than 1 streamdeck? 
all get the same config?

**Reply by flcou** [reply]
Yes, that would be the idea.

**Comment by Narcodepsy** [comment]
This would remove the ability for multiple steamdecks to be used on the
same system with different configs.

Do the import export function not fit the ability to quickly make multiple
streamdecks have the same config if desired?

**Comment by remdzi** [comment]
I work with flocou on the same project.

The idea is to support both use cases.

If the serial number is specified in the json
```json
{
    "streamdeck_ui_version": 1,
    "state": {
        "AL42K2C55543": {
            "buttons": {
```
then the configuration shall apply to this specific streamaeck.
This is the current behavior of streamdeck-ui and we do not want to change it.

But if the serial number is unspecified (or maybe a special character is used like '*')
```json
{
    "streamdeck_ui_version": 1,
    "state": {
        "*": {
            "buttons": {
```
then the configuration shall applies to any connected streamdeck.
This is the new feature we need.

We need this use case to be able to replace a broken, stolen or lost streamdeck on our machine without the need for any reconfiguration.
As you do, for example, when you change your keyboard on your computer. Just plug and play.

For use case with multiple streamdecks with different configurations we agree this is not possible.
But let's support the most simple, and most probably also common use case when a single streamdeck is used.

**Reply by coolapso** [reply]
yeah I could definitely see a scenario and usage like that, even go a bit further.

Default configuration: Applies to all devices 
Single Device configuration: Applies only to single device
Apply to multiple devices: Applies a given configuration to the specified streamdecks serial ids, so for example if someone has 4 streamdecks, 2 can have the default and 2 others can have some other configuration applied to both. 


Ill be happy to review any PR coming out for this feature.

**Comment by sftim** [comment]
Another option: allow adding a new Stream Deck and cloning its button configuration from a different Stream Deck.

That will also help if you (eg) replace a v1 Stream Deck with a v42 Stream Deck, and the configuration isn't directly portable. The clone button could be responsible for adapting the settings.

**Comment by rkmax** [comment]
Maybe we can implement the concept of "profile" and basically store everything we already have in the profiles instead of devices.

With that we can implement the following features:

- profiles have names and can be renamed 
- import/export profiles
- see all profiles
- select profile for selected device
- switch profile based on the current active application
- each device connected will have a new profile (same device name) by default to preserve behavior or empty for new devices

**Comment by inkhey** [comment]
Hello all :wave: , i made a proposal for the "default configuration part" of this issue : https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/pull/87 .
